unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/DefaultChatEntry.cs

```diff
@@ -111,18 +112,13 @@ private string GetUserString(ChatEntryModel chatEntryModel)
 
         private string GetCoordinatesLink(string body)
         {
-            if (!CoordinateUtils.HasValidTextCoordinates(body))
-                return body;
-            var textCoordinates = CoordinateUtils.GetTextCoordinates(body);
+            if (!CoordinateUtils.HasValidTextCoordinates(body)) return body;
 
-            for (var i = 0; i < textCoordinates.Count; i++)
-            {
-                // TODO: the preload should not be here
-                PreloadSceneMetadata(CoordinateUtils.ParseCoordinatesString(textCoordinates[i]));
+            List<string> textCoordinates = CoordinateUtils.GetTextCoordinates(body);
 
-                body = body.Replace(textCoordinates[i],
-                    $"</noparse><link={textCoordinates[i]}><color=#4886E3><u>{textCoordinates[i]}</u></color></link><noparse>");
-            }
+            foreach (string coordinates in textCoordinates)
```

Comment:
Did change it, but not sure if complies to the request. Can you check?

Reply:
Not quite
This is what I meant: 
https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.replace?view=net-7.0#system-text-regularexpressions-regex-replace(system-string-system-text-regularexpressions-matchevaluator)

so instead of `GetTextCoordinates` you make `Replace` by `REGEX` itself in a single pass: now you collect `Matches` and then process them, with the proposed change you will `Match` and `Replace` in the same call

Reply:
Done, much better now